### ISSUE TITLE
Albums/Labels: Option to mark albums and  labels as private or archive

### ISSUE THREAD

**Issue by tehniemer** [photoprism/photoprism#480]
It would be nice to be able to have an album or folder level privacy setting so that all files in the folder or album are set to private at once without having to select every item in the folder. Ideally it would also set any new items added to that folder or album in the future to private without any additional user interaction. 

------------------------------------------------
Acceptance Criteria:

* The user MUST be able to mark albums (albums, moments, states, calendar, folders) and labels as private or archived
* All items that are added to an album or label marked as private or archived MUST be private/archived as well

See: https://github.com/photoprism/photoprism/issues/1528

TBD: What happens if a user tries to remove the private/archive flag from a photo that is part of an private album?

**Comment by pluja** [comment]
I think this would be a nice-to-have and a very handy feature. Also, the ability of having private albums, just like you can have private photos would be nice.

**Comment by Ayo91** [comment]
Yep, I miss this feature too.. I have complete albums that I would like to convert to private. There is not option for "select all" photos so, there is not any alternative

**Comment by Code-nomad** [comment]
Same

**Comment by gu33mis** [comment]
Agreed, I really miss ability to have an album or folder level privacy setting. It would be great if all photos added to the album with the privacy setting ON could inherit privacy setting instead of changing it manually for each individual file in the album.

**Comment by lastzero** [comment]
We started experimenting with this today. Automatically flagging all pictures in an album as private is tricky in that most album types are basically just search filters. These filters can match a large number of pictures or even everything.

However, flagging albums as private (as implemented today) seems to be useful when there are roles with limited permissions so that some users can't see them. Since you can also disable search, this COULD effectively prevent users from seeing the content without flagging all pictures in an album as private.

Because this use case depends on the availability of additional features that have not yet been released, I have added a feature flag to hide the "Private" switch until it can be used in a meaningful way. It should also be noted that the flags affect the number of visible albums, so this adds some complexity to the frontend and backend as well (unless you are okay with incorrect counts being displayed in the navigation).

**Comment by Morgan-6Freedom** [comment]
Is it implemented ?

**Comment by lastzero** [comment]
@Morgan-6Freedom The private flag for albums should work for Plus members if they have experimental features enabled. So it's not officially released yet as we want to make sure it's implemented in a useful way. Feel free to email us if you need access.

**Comment by rob-rocks** [comment]
Would it be possible to mark specific directories as private, i.e. similar to the .ppignore file? I.e. for folders that contain only private pictures like for instance a smartphone upload folder

**Comment by ExNought** [comment]
I want to add to that. I want to separate some pictures explicitly and first uploading them, then waiting for indexing and then selecting and privatizing seems like a bad idea in some cases :)

Think of the children maybe ;)

**Comment by venik-fi** [comment]
> Would it be possible to mark specific directories as private, i.e. similar to the .ppignore file? I.e. for folders that contain only private pictures like for instance a smartphone upload folder

Hello, I understand this is a very old idea.. But still we agree to be a very useful feature.
 Any news here?
 Thank you in advance

**Comment by Brent-Chiono** [comment]
Here we are again 5 years later at the end of 2025. I have been waiting for this feature for a long time. I understand it is open source software but I think privacy should jump up the list a little bit.
